encodings/sparse/src/canonical.rs

```diff
@@ -75,6 +75,12 @@ fn canonicalize_sparse_primitives<
         )
     };
 
+    // If the number of values in the sparse array matches the array length, then all
+    // values are in fact patches, since patches are sorted this is the correct values.
+    if patches.array_len() == patches.values().len() {
+        return patches.values().to_canonical();
+    }
```

Comment:
when do we end up with this? sparse mostly-null and then we `filter` to non-null or something? should one of the sparse compute fns be returning a different encoding more eagerly?

Reply:
Updated slice and take